docs/en/interfaces/cli.md

```diff
@@ -158,6 +158,116 @@ $ clickhouse-client --param_tuple_in_tuple="(10, ('dt', 10))" -q "SELECT * FROM
 $ clickhouse-client --param_tbl="numbers" --param_db="system" --param_col="number" --query "SELECT {col:Identifier} FROM {db:Identifier}.{tbl:Identifier} LIMIT 10"
 ```
 
+## Connection string {#connection_string}
+
+The connection string for clickhouse-client is presented in URI format:
+
+```text
+clickhouse://[user_info@][hosts_and_ports][/dbname][?query_parameters]
+```
+
+where user_info is: `user[:password]`
+and hosts_and_ports is a list of values: `[host][:port],[host][:port]` Port is not mandatory.
+and query_parameters is a list of parameter[=value]: `param_name[=value]&param_name[=value]...` value may not be required for some of the parameters. Parameter names are case sensitive.
+
+Allowed query_parameters keys:
+
+- `secure` or shorthanded `s` - no value. If specified, client will connect to the server over a secure connection (TLS). See `secure` in [command-line-options](#command-line-options)
```

Comment:
Ah, interesting. Sounds like only `s`/`secure` is allowed as query parameter. Should that not accept pretty much all options (in particular session settings) that can be passed as of now to clickhouse-client?

Reply:
We've been thinking about this question.

And we decided that the connection string will contain only the parameters that are related to the connection.
In other databases, the same restrictions apply. 

I think we would need these parameters added later:
- timeout <seconds>
- retry count <count>

Reply:
Makes sense.

(if you have links to similar parameters in other DBMS, please share them - it would be interesting to read about them).

Reply:
Other DBMS also have common several parameters related to SSL: `certificate`, `password to client certificate`, `root  Certificate Authority`.

We can't set those this in clickhouse-client command line interface, but it is allowed to set some parameters in clickhouse-client XML configuration:
https://clickhouse.com/docs/en/interfaces/cli#configuration_files